### PR TITLE
Migrate configuration to Pydantic settings

### LINKED ISSUE #1315: Migrate configs to the pydantic-settings
Refactor the configuration management within the SKALE admin docker container to use [pydantic-settings](https://docs.pydantic.dev/latest/concepts/pydantic_settings/) instead of the current approach. This migration will improve configuration validation, type-safety, and maintainability.

**Acceptance Criteria:**
- Identify all configuration classes, files, and environment variable handling code.
- Refactor relevant components to use `pydantic-settings` for settings management.
- Update documentation and usage examples to match the new approach.
- Ensure all tests pass and add new ones for any additional edge-cases.

**Benefits:**
- Enhanced validation and clearer config structure
- Easier extension for new settings
- Better error reporting and debugging for config issues

### PR DESCRIPTION
**Pull request by alexgex**
This pull request introduces several important updates to improve configuration management, time handling, and logging throughout the codebase. The main changes include centralizing configuration using the `SkaleSettings` class, enforcing timezone-aware datetimes, updating logging to better handle sensitive information, and updating dependencies and test fixtures for better consistency and reliability.

**Configuration and Initialization Improvements:**

* Refactored configuration management to use the `SkaleSettings` class, updating initialization of `BountyAgent`, `init_skale`, and related functions to accept a `settings` parameter, and centralized settings file paths in `configs/__init__.py` [[1]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL59-R69) [[2]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL182-R194) [[3]](diffhunk://#diff-d01f411a2f32baa2bf1b181dccc66d4571af1c85cf38f05967396e348e026dc3R2-R17) [[4]](diffhunk://#diff-d01f411a2f32baa2bf1b181dccc66d4571af1c85cf38f05967396e348e026dc3L25-R33) [[5]](diffhunk://#diff-3984f02a262573812c5a6471a9e6d96a567cb1e0b0fc9675b9d7635f01588e87R29) [[6]](diffhunk://#diff-3984f02a262573812c5a6471a9e6d96a567cb1e0b0fc9675b9d7635f01588e87L39-L42) [[7]](diffhunk://#diff-3984f02a262573812c5a6471a9e6d96a567cb1e0b0fc9675b9d7635f01588e87L53-R65).
* Updated test fixtures to provide and use `SkaleSettings`, ensuring tests use consistent configuration and environment variables [[1]](diffhunk://#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R7-R18) [[2]](diffhunk://#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R27-R58) [[3]](diffhunk://#diff-9c743658cf40372b99ab39e8772123e88863501073cd22b3b691384d700f4e9bL43-R44) [[4]](diffhunk://#diff-616cb4dc73142fc11062ca281a0225d5e44423a605ca9cec376f4acfe75fda0fL29-R59).

**Timezone and Date Handling:**

* Standardized all datetime usage to be timezone-aware and in UTC, replacing `datetime.utcnow()` and `datetime.utcfromtimestamp()` with `datetime.now(timezone.utc)` and `datetime.fromtimestamp(..., timezone.utc)` across the codebase and tests [[1]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL28-R33) [[2]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL93-R100) [[3]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL137-R144) [[4]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL148-R155) [[5]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL159-R166) [[6]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL168-R175) [[7]](diffhunk://#diff-9c743658cf40372b99ab39e8772123e88863501073cd22b3b691384d700f4e9bL57-R57) [[8]](diffhunk://#diff-9c743658cf40372b99ab39e8772123e88863501073cd22b3b691384d700f4e9bL77-R77) [[9]](diffhunk://#diff-9c743658cf40372b99ab39e8772123e88863501073cd22b3b691384d700f4e9bL104-R111).

**Logging and Sensitive Data Handling:**

* Enhanced the logging system to pass `sgx_url` and `endpoint` into file handlers, and updated the `compose_hiding_patterns` function to dynamically hide sensitive information based on these values [[1]](diffhunk://#diff-6c951f7291399b1ac2d7f64a72e30ba452ea670eae38ffb720736be7a0a53fbbL59-R69) [[2]](diffhunk://#diff-f95d61f92ec1928f16e5f12db8a965c0845a8a56682f121f8b2cb3b56ecd058eL26-R45) [[3]](diffhunk://#diff-f95d61f92ec1928f16e5f12db8a965c0845a8a56682f121f8b2cb3b56ecd058eL67-R73).

**Dependency and Environment Updates:**

* Upgraded the `skale.py` dependency to version `7.12dev2` for improved compatibility and features.
* Added `DEFAULT_GAS_PRICE_WEI` to the environment variables and ensured low gas price is used in tests to prevent insufficient funds errors [[1]](diffhunk://#diff-7077afefe0e1d21088fb33b718696c1b30cdaa396609d6f8b46174c7a0fd1b04R10) [[2]](diffhunk://#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R27-R58).

**Test and Helper Function Updates:**

* Updated helper and test functions to work with the new configuration system, and fixed method calls to align with the latest `SkaleManager` interface [[1]](diffhunk://#diff-3984f02a262573812c5a6471a9e6d96a567cb1e0b0fc9675b9d7635f01588e87L78-R76) [[2]](diffhunk://#diff-9c743658cf40372b99ab39e8772123e88863501073cd22b3b691384d700f4e9bL43-R44) [[3]](diffhunk://#diff-616cb4dc73142fc11062ca281a0225d5e44423a605ca9cec376f4acfe75fda0fL29-R59).

These changes collectively improve maintainability, reliability, and security in both production and testing environments.

### CODE REVIEW

File: tools/helper.py [hunk 1]
```diff
@@ -36,10 +37,7 @@
     NOTIFIER_URL,
     REDIS_URI,
     SGX_CERTIFICATES_FOLDER,
-    SGX_SERVER_URL,
-    STATE_FILEPATH,
 )
-from configs.web3 import ENDPOINT, MANAGER_CONTRACTS
 from tools.exceptions import NodeNotFoundException
```
Comment (dmytrotkk) [inline tools/helper.py:41]: why `configs.web3` module is still present in the repo?

**Review by dmytrotkk — APPROVED** [review]
